File: abc/abc242_c.py
N = int(input())
MOD = 998244353

a = b = c = d = e = 1
for i in range(N - 1):
    a, b, c, d, e = a + b, a + b + c, b + c + d, c + d + e, d * 2 + e
    if i % 16 == 0:
        a %= MOD
        b %= MOD
        c %= MOD
        d %= MOD
        e %= MOD

print(((a + b + c + d) * 2 + e) % MOD)

File: abc/abc242_c.cpp
#include <iostream>
#include <vector>
using namespace std;

int main()
{
    int N;
    cin >> N;
    const int MOD = 998244353;
    vector<int> dp(9, 1);
    for (int i = 0; i < N - 1; i++)
    {
        vector<int> ndp(9);
        for (int j = 0; j < 9; j++)
        {
            if (j > 0)
            {
                ndp[j] = (ndp[j] + dp[j - 1]) % MOD;
            }
            ndp[j] = (ndp[j] + dp[j]) % MOD;
            if (j < 9)
            {
                ndp[j] = (ndp[j] + dp[j + 1]) % MOD;
            }
        }
        for (int j = 0; j < 9; j++)
        {
            dp[j] = ndp[j] % MOD;
        }
    }

    int ans = 0;
    for (int i = 0; i < 9; i++)
    {
        ans += dp[i];
        ans %= MOD;
    }
    cout << ans << endl;
}
